flytepropeller/pkg/compiler/validators/typing_test.go

```diff
@@ -893,3 +893,10 @@ func TestStructuredDatasetCasting(t *testing.T) {
 		assert.True(t, castable, "StructuredDataset are nullable")
 	})
 }
+
+// TODO add tests for blob typechecker
+//
+// WIP test suite
+// assert.True(t, castable, "Blob(format='csv') should be castable to Blob()")
+// assert.True(t, castable, "Blob() should be castable to Blob()")
+// assert.False(t, castable, "Blob(format='csv') should not be castable to Blob('zip')")
```

Comment:
note-to-self: understand this test file and understand how typechecker works in order to define the correct spec for the new blob typechecker